### PR TITLE
Story2

### PR DESCRIPTION
**Pull request by shellycs50**

### CODE REVIEW

File: sarcasticfringeheadapi/.env.example [hunk 1]
```diff
@@ -1,6 +1,6 @@
 APP_NAME=Laravel
 APP_ENV=local
-APP_KEY=
+APP_KEY=base64:ojmIewug9EIVSlltKrQVyp3By9+nrVS2yB/37qBl6A8
```
Comment (mporam) [inline sarcasticfringeheadapi/.env.example]: this should be in the actual env file, not the example one

Comment (mporam) [reply sarcasticfringeheadapi/.env.example]: the whole point of the example one is it doesnt have live data in it

File: sarcasticfringeheadapi/app/Http/Controllers/BookController.php [hunk 2]
```diff
@@ -13,8 +13,121 @@ public function __construct(Book $book)
         $this->book = $book;
     }
 
-    public function getAllBooks()
+    private function serialize($books)
     {
-        // ** TODO ** 
+        $output = [];
+        foreach ($books as $book) {
+            $item = [
```
Comment (mporam) [inline sarcasticfringeheadapi/app/Http/Controllers/BookController.php]: why create item and not just assign to `$output[]`

File: sarcasticfringeheadapi/app/Http/Controllers/BookController.php [hunk 3]
```diff
@@ -13,8 +13,121 @@ public function __construct(Book $book)
         $this->book = $book;
     }
 
-    public function getAllBooks()
+    private function serialize($books)
```
Comment (mporam) [inline sarcasticfringeheadapi/app/Http/Controllers/BookController.php]: missing type hints

File: sarcasticfringeheadapi/app/Http/Controllers/BookController.php [hunk 4]
```diff
@@ -13,8 +13,121 @@ public function __construct(Book $book)
...
+            ];
+            $output[] = $item;
     }
+    return $output;
 }
+
+    public function getAllBooks(Request $request)
+    {
+        $books = $this->book->query();
+        $books = $books->get();
```
Comment (mporam) [inline sarcasticfringeheadapi/app/Http/Controllers/BookController.php]: why not `$this->books->all()` ?

File: sarcasticfringeheadapi/app/Http/Controllers/BookController.php [hunk 5]
```diff
@@ -13,8 +13,121 @@ public function __construct(Book $book)
...
+        foreach($books as $book)
+        {
+            $book->genre;
+        }
+        $books = $this->serialize($books);
```
Comment (mporam) [inline sarcasticfringeheadapi/app/Http/Controllers/BookController.php]: better to create a new variable than assign over the top of an existing one. create something more semantic like `$serializedBooks`

File: sarcasticfringeheadapi/app/Http/Controllers/BookController.php [hunk 6]
```diff
@@ -13,8 +13,121 @@ public function __construct(Book $book)
...
+        if (count($books) === 0)
+        {
+            return response()->json([
+                'message' => 'No Books Found'
+            ], 404);
+        }
+        return response()->json(["message" => "Books successfully retrieved", "data" => $books], 200);
+        //may need to edit $books variable to delete multiple instances of genre_id if not handled by laravel automatically. (because book table has genre_id and genre table has id)
```
Comment (mporam) [inline sarcasticfringeheadapi/app/Http/Controllers/BookController.php]: should probably go check that then

File: sarcasticfringeheadapi/app/Http/Controllers/BookController.php [hunk 7]
```diff
@@ -13,8 +13,121 @@ public function __construct(Book $book)
...
+    public function getSingleBookById (Request $request, $id)
+    {
+        $book = $this->book->find($id);
+        if (!$book) {
+            return response()->json([
+                'message' => "Sorry, that book does not exist"
+            ], 404);
+        }
+        $book->genre;
+        $book = $this->serialize([$book]);
```
Comment (mporam) [inline sarcasticfringeheadapi/app/Http/Controllers/BookController.php]: same as above, make a new variable

File: sarcasticfringeheadapi/app/Http/Controllers/BookController.php [hunk 8]
```diff
@@ -13,8 +13,121 @@ public function __construct(Book $book)
...
+
+
+
+
+
+
+
+
+
+
```
Comment (mporam) [inline sarcasticfringeheadapi/app/Http/Controllers/BookController.php]: why?!

File: sarcasticfringeheadapi/app/Http/Controllers/BookController.php [hunk 9]
```diff
@@ -13,8 +13,121 @@ public function __construct(Book $book)
...
+    /*
+    ** dont use yet. part of later stories. **
```
Comment (mporam) [inline sarcasticfringeheadapi/app/Http/Controllers/BookController.php]: if you dont need it, delete it

**Review by mporam — CHANGES_REQUESTED** [review]

File: sarcasticfringeheadapi/app/Http/Controllers/BookController.php (hunk 4 above)
Comment (shellycs50) [reply sarcasticfringeheadapi/app/Http/Controllers/BookController.php:39]: because its going to need to be a query later (tomorrow)

**Pull request merged**